### PR TITLE
custom paymaster + smart account config

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: packages/account-abstraction/src/BiconomyPaymasterAPI.ts [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
-  // Might maintain API key at smart account level
-  constructor(
-    readonly apiUrl: string,
-    readonly dappAPIKey: string,
-    readonly payMasterAddress: string
-  ) {
+import { IPaymasterAPI } from '@biconomy-sdk/core-types'
+
+export class BiconomyPaymasterAPI implements IPaymasterAPI {
+  constructor(readonly apiUrl: string, readonly dappAPIKey: string) {
```
Comment (tomarsachin2271) [inline packages/account-abstraction/src/BiconomyPaymasterAPI.ts:7]: apiUrl should be renamed to signingServiceUrl

File: packages/core-types/src/smart-account.types.ts [hunk 2]
```diff
@@ -9,19 +9,21 @@ import { MultiSendCallOnlyContract } from './contracts/MultiSendCallOnlyContract
...
   signType: SignTypeMethod
-  providerUrlConfig?: ProviderUrlConfig[]
+  // dappAPIKey?: string
+  networkConfig: NetworkConfig[]
   entryPointAddress?: string
-  bundlerUrl?: string
-  paymasterAddress?: string
-  signingServiceUrl: string
+  biconomySigningServiceUrl?: string
+  // bundlerUrl?: string
```
Comment (tomarsachin2271) [inline packages/core-types/src/smart-account.types.ts:24]: Uncomment bundlerUrl here

File: packages/core-types/src/smart-account.types.ts [hunk 3]
```diff
@@ -9,19 +9,21 @@ import { MultiSendCallOnlyContract } from './contracts/MultiSendCallOnlyContract
 import { SmartWalletContract } from './contracts/SmartWalletContract'
 import { GasLimit } from './transaction.types'
 import { Signer } from 'ethers'
+import { IPaymasterAPI } from 'account-abstraction-types'
 
 export interface SmartAccountConfig {
   activeNetworkId: ChainId // same
   supportedNetworksIds: ChainId[] // Network[] chainId: CbainId, rpcUrl?: string
```
Comment (tomarsachin2271) [inline packages/core-types/src/smart-account.types.ts:16]: Add environment field which can have values "testnet" or "mainnet" and maintain two default values, 1 for testnet and 1 for mainnet. Be default it should be "mainnet"

Comment (livingrockrises) [reply packages/core-types/src/smart-account.types.ts:16]: can you share more details? i missed this part of the discussion

File: packages/core-types/src/smart-account.types.ts (hunk 2 above)
Comment (livingrockrises) [reply packages/core-types/src/smart-account.types.ts]: done

File: packages/account-abstraction/src/BiconomyPaymasterAPI.ts (hunk 1 above)
Comment (livingrockrises) [reply packages/account-abstraction/src/BiconomyPaymasterAPI.ts:7]: done

File: packages/core-types/src/smart-account.types.ts (hunk 3 above)
Comment (tomarsachin2271) [reply packages/core-types/src/smart-account.types.ts]: The default values for the config which we have are of test networks but we should have one default configuration for main networks also.... and to do that we can have one 'environment' field in SmartAccountConfig which can have values "test" or "mainnet".. 

Like we have for [Hyphen](https://docs.biconomy.io/products/hyphen-instant-cross-chain-transfers/sdk#2.-importing-and-instantiation)

So if developer set this field as "test" then we'll use the defaultConfig value which has test networks values.. and vice versa.

Comment (livingrockrises) [reply packages/core-types/src/smart-account.types.ts]: got it. let me create another PR

**Pull request closed without merging**